### PR TITLE
Update gnostic library to point to gnostic-models

### PR DESCRIPTION
**Pull request by Jefftree**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

gnostic OpenAPI components were moved to the gnostic-models library. gnostic is pinned on [v0.5.7-v3refs](https://github.com/kubernetes/kube-openapi/blob/master/go.mod#L12) because we did not want to capture the additional dependencies introduced with later versions of gnostic. The OpenAPI component has been moved to gnostic-models so update the library path to reflect that.

This PR also bumps kustomize and kube-openapi to the latest version.

It picks up https://github.com/kubernetes/kube-openapi/pull/403 for a thread safe openapi cache.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
OpenAPI proto deserializations should use gnostic-models instead of the gnostic library
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/triage accepted
/assign @apelisse @sttts

### CODE REVIEW

**Comment by Jefftree** [comment]
pull-kubernetes-node-e2e-containerd-dra has failed but it looks broken https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-node-e2e-containerd-dra and is not a required test for merge.

**Comment by sttts** [comment]
/lgtm
/approve

Pure dependency update.

**Comment by dims** [comment]
the dependency stats are moving in the right direction

```
--- /logs/artifacts/stats-base.txt	2023-06-02 14:38:22.509602104 +0000
+++ /logs/artifacts/stats.txt	2023-06-02 14:38:22.057567358 +0000
@@ -1,6 +1,6 @@
 Direct Dependencies: 210 
-Transitive Dependencies: 342 
-Total Dependencies: 3[9](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/118384/check-dependency-stats/1664641754902564864#1:build-log.txt%3A9)1 
+Transitive Dependencies: 340 
+Total Dependencies: 389 
 Max Depth Of Dependencies: 17 
 All dependencies:
 
@@ -[18](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/118384/check-dependency-stats/1664641754902564864#1:build-log.txt%3A18)9,7 +189,6 @@
 github.com/docker/docker
 github.com/docker/go-connections
 github.com/docker/go-units
-github.com/docopt/docopt-go
 github.com/dougm/pretty
 github.com/dustin/go-humanize
 github.com/emicklei/go-restful/v3
@@ -[23](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/118384/check-dependency-stats/1664641754902564864#1:build-log.txt%3A23)2,7 +231,7 @@
 github.com/google/btree
 github.com/google/cadvisor
 github.com/google/cel-go
-github.com/google/gnostic
+github.com/google/gnostic-models
 github.com/google/go-cmp
 github.com/google/gofuzz
 github.com/google/pprof
@@ -270,7 +[26](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/118384/check-dependency-stats/1664641754902564864#1:build-log.txt%3A26)9,6 @@
 github.com/matttproud/golang_protobuf_extensions
 github.com/mistifyio/go-zfs
 github.com/mitchellh/go-wordwrap
-github.com/mitchellh/mapstructure
 github.com/moby/ipvs
 github.com/moby/spdystream
 github.com/moby/sys/mountinfo
```

**Comment by apelisse** [comment]
Yes, we basically just moved the file from `github.com/google/gnostic/openapiv2` to `github.com/google/gnostic-models/openapiv2` as mentioned in the error that you have. 

I'm curious why this fails @Jefftree , is it because client-go doesn't use a properly pinned version of kube-openapi?

**Comment by imjasonh** [comment]
@apelisse Is there automation in place to keep that repo's dependencies up to date? I see a few available dep updates if I `go get -u`. I also don't see any presubmit testing on changes to that repo, which is a little worrying.

File: vendor/github.com/evanphx/json-patch/patch.go [hunk 1]
```diff
@@ -568,29 +568,6 @@ func (p Patch) replace(doc *container, op Operation) error {
 		return errors.Wrapf(err, "replace operation failed to decode path")
 	}
 
-	if path == "" {
```
Comment (liggitt) [inline vendor/github.com/evanphx/json-patch/patch.go:571]: it looks like this lost this fix: https://github.com/evanphx/json-patch/commit/50fdc0b4c9ab36933e6a0c9288fef3ac5df2b907

File: go.mod [hunk 2]
```diff
@@ -29,7 +29,7 @@ require (
 	github.com/docker/distribution v2.8.2+incompatible
 	github.com/docker/go-units v0.5.0
 	github.com/emicklei/go-restful/v3 v3.9.0
-	github.com/evanphx/json-patch v4.12.0+incompatible
+	github.com/evanphx/json-patch v5.6.0+incompatible
```
Comment (liggitt) [inline go.mod:32]: `v5.6.0+incompatible` is *older* than `v4.12.0+incompatible` ... this dropped the following fix:

https://github.com/evanphx/json-patch/compare/v5.6.0...v4.12.0

do we care about this?

**Pull request merged**